neo/Consensus/ConsensusContext.cs

```diff
@@ -417,37 +412,27 @@ public void Serialize(BinaryWriter writer)
             writer.Write(TransactionHashes ?? new UInt256[0]);
             writer.Write(Transactions?.Values.ToArray() ?? new Transaction[0]);
             writer.WriteVarInt(PreparationPayloads.Length);
-            foreach (var payload in PreparationPayloads)
-            {
-                bool hasPayload = !(payload is null);
-                writer.Write(hasPayload);
-                if (!hasPayload) continue;
-                writer.Write(payload);
-            }
-            writer.WriteVarInt(CommitPayloads.Length);
-            foreach (var payload in CommitPayloads)
-            {
-                bool hasPayload = !(payload is null);
-                writer.Write(hasPayload);
-                if (!hasPayload) continue;
-                writer.Write(payload);
-            }
-            writer.WriteVarInt(ChangeViewPayloads.Length);
-            foreach (var payload in ChangeViewPayloads)
-            {
-                bool hasPayload = !(payload is null);
-                writer.Write(hasPayload);
-                if (!hasPayload) continue;
-                writer.Write(payload);
-            }
-            writer.WriteVarInt(LastChangeViewPayloads.Length);
-            foreach (var payload in LastChangeViewPayloads)
+            SerializeConsensusPayloadArray(writer, PreparationPayloads);
+            writer.WriteVarInt(Validators.Length);
+            SerializeConsensusPayloadArray(writer, CommitPayloads);
+            SerializeConsensusPayloadArray(writer, ChangeViewPayloads);
+            SerializeConsensusPayloadArray(writer, LastChangeViewPayloads);
+        }
+        private void SerializeConsensusPayloadArray(BinaryWriter writer, ConsensusPayload[] payloadArray)
+        {
+            foreach (var payload in payloadArray)
             {
                 bool hasPayload = !(payload is null);
                 writer.Write(hasPayload);
                 if (!hasPayload) continue;
                 writer.Write(payload);
             }
         }
+        private void ReadConsensusPayloadArray(BinaryReader reader, out ConsensusPayload[] payloadArray, ulong payloadLenght)
```

Comment:
Why use output parameter instead of return value?

Reply:
Let's change

Reply:
Done.